File: java/sprint5/A/Solution.java
public class Solution {
    public static int treeSolution(Node head) {
        // Your code
        // “ヽ(´▽｀)ノ”
    }

    // <template>
    private static class Node {
        int value;  
        Node left;  
        Node right;  
    
        Node(int value) {  
            this.value = value;
            this.left = null;
            this.right = null;
        }
    }
    // <template>


    private static void test() {
        Node node1 = new Node(1);
        Node node2 = new Node(-5);
        Node node3 = new Node(3);
        node3.left = node1;
        node3.right = node2;
        Node node4 = new Node(2);
        node4.left = node3;
        assert treeSolution(node4) == 3;
    }
}

File: kotlin/sprint5/A/Solution.kt
// <template>
class Node(var value: Int) {
    var left: Node? = null
    var right: Node? = null
}
// <template>

fun treeSolution(head: Node?): Int {
    // Your code
    // “ヽ(´▽｀)ノ”
}

fun test() {
    val node1 = Node(1)
    val node2 = Node(-5)
    val node3 = Node(3)
    node3?.left = node1
    node3?.right = node2
    val node4 = Node(2)
    node4?.left = node3
    assert(treeSolution(node4) == 3)
}